server/channels/jobs/migrations/worker.go

```diff
@@ -100,10 +100,10 @@ func (worker *Worker) DoJob(job *model.Job) {
 		return
 	}
 
-	cancelContext := request.EmptyContext(worker.logger)
+	var cancelContext request.CTX = request.EmptyContext(worker.logger)
```

Comment:
Can we just make `WithContext` return the concrete type instead of `request.CTX`? Since `request.Context` already implements that interface, it can be used anywhere that a `request.CTX` is expected. It also seems not idiomatic to have the "constructor" follow the "accept interfaces, return structs" paradigm, but have `Clone` and the sibling methods return an interface from a concrete type.

Reply:
It's complicated. It's possible to change `Context.WithContext(ctx context.Context) *CTX` to `Context.WithContext(ctx context.Context) *Context`. But that would also require changing the signature method of `CTX.WithContext`. See https://goplay.tools/snippet/jh-xx6H5K47. That is a path that I'm hesitant to go down, as it would tear down the boundary between the interface and the implementation.